include/seqan3/io/alignment_file/format_bam.hpp

```diff
@@ -56,63 +56,15 @@ namespace seqan3
  *
  * \copydetails seqan3::format_sam
  */
-struct format_bam
+struct format_bam : private detail::format_sam_base
```

Comment:
private ineheritance o_O ?

Also I would make this a class now that it has stuff like inheritance.

Reply:
should the other file formats become classes to? They have a lof of member function but not custom constructors nor inheritance

Reply:
Hm, an edge case I would say. Up to you.